Save commit

NikolayAbysov opened the pull request.

NikolayAbysov committed:
Save commit

NikolayAbysov committed:
Solution implemented

NikolayAbysov committed:
Solution implemented

RomanDubovskyi commented on src/main/java/mate/academy/internetshop/dao/impl/UserDaoImpl.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public User update(User user) {
+        Storage.users.set(Storage.users.indexOf(user), user);
+        return null;
```
тут потрібно завжди null повертати?

NikolayAbysov replied:
Yep, that's really strange:)

RomanDubovskyi commented on src/main/java/mate/academy/internetshop/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,46 @@
+    public List<Order> getUserOrders(User user) {
+        return orderDao.getAll()
+                .stream()
+                .filter(o -> o.getUser().equals(user))
```
чи корректно відпрацює порівняння якщо у нас не перевизначений equals для User?
btw, ти можеш використати для цього id юзера

NikolayAbysov replied:
My mistake. Sure, we have to use id, cause we have basic Object equals() method.

RomanDubovskyi commented on src/main/java/mate/academy/internetshop/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Override
+    public Order completeOrder(List<Product> products, User user) {
+        return orderDao.create(new Order(products, user));
```
давай одразу тут будемо очищати корзину юзера

NikolayAbysov committed:
Issues fixed:
1 User update() returns User object instead of null;
2 OrderServiceImpl getUserOrders() changed to equals by id
3 OrderServiceImpl completeOrder() functionality of cleaning ShopCart added

boroda4436 commented on src/main/java/mate/academy/internetshop/Application.java:
```diff
@@ -21,7 +25,17 @@ public static void main(String[] args) {
+        System.out.println(userService.delete(user.getId()));
+
+        ShoppingCart shoppingCart = new ShoppingCart();
+        ShoppingCartService shoppingCartService
```
provide more examples of ShoppingCartService usage

NikolayAbysov replied:
@boroda4436 
Hi!

Additional examples and fixes added :)

NikolayAbysov committed:
Additional cases of ShoppingCartService usage provided.

Bugs fixed:

1 Init of ArrayList of Products added to ShoppingCart
2 Check of ShoppingCar on null Id added in addProduct method
in ShoppingCartService
